pkg/seedadmissioncontroller/webhooks/admission/extensionresources/admission.go

```diff
@@ -0,0 +1,282 @@
+				return validation.ValidateOperatingSystemConfigUpdate(new, old)
+			}})
+
+	h.addResource(metav1.GroupVersionResource{Group: extensionsv1alpha1.SchemeGroupVersion.Group, Version: extensionsv1alpha1.SchemeGroupVersion.Version, Resource: "workers"},
+		artifact{
+			newEntity: func() client.Object { return new(extensionsv1alpha1.Worker) },
+			validateCreateResource: func(n, _ client.Object) field.ErrorList {
+				new := n.(*extensionsv1alpha1.Worker)
+				return validation.ValidateWorker(new)
+			},
+			validateUpdateResource: func(n, o client.Object) field.ErrorList {
+				new := n.(*extensionsv1alpha1.Worker)
+				old := o.(*extensionsv1alpha1.Worker)
+				return validation.ValidateWorkerUpdate(new, old)
+			}})
+
+	return &h
+}
+
+type handler struct {
+	decoder   *admission.Decoder
+	logger    logr.Logger
+	artifacts map[metav1.GroupVersionResource]artifact
+}
+
+func (h *handler) addResource(gvr metav1.GroupVersionResource, art artifact) {
+	h.artifacts[gvr] = art
+}
+
+var _ admission.Handler = &handler{}
+
+func (h *handler) InjectDecoder(d *admission.Decoder) error {
+	h.decoder = d
+	return nil
+}
+
+func (h *handler) Handle(ctx context.Context, ar admission.Request) admission.Response {
+	h.logger.Info(fmt.Sprintf("validating resource of type %s for operation %s", ar.Resource, ar.Operation))
+
+	_, cancel := context.WithTimeout(ctx, 10*time.Second)
+	defer cancel()
+
+	artifact, ok := h.artifacts[ar.Resource]
+	if !ok {
+		return admission.Allowed("validation not found for the given resource")
+	}
+
+	switch ar.Operation {
+	case admissionv1.Create:
+		return h.handleValidation(ar.Object, ar.OldObject, artifact.newEntity, artifact.validateCreateResource)
+	case admissionv1.Update:
+		return h.handleValidation(ar.Object, ar.OldObject, artifact.newEntity, artifact.validateUpdateResource)
+	default:
+		return admission.Allowed("operation is not CREATE or UPDATE")
+	}
+}
+
+type newEntity func() client.Object
+type validate func(new, old client.Object) field.ErrorList
+
+// artifact servers as a helper to setup the corresponding function.
+type artifact struct {
+	// newEntity is a simple function that creates and returns a new resource.
+	newEntity newEntity
+
+	// validateCreateResource is a wrapper function for the different create validation functions.
+	validateCreateResource validate
+
+	// validateUpdateResource is a wrapper function for the different update validation functions.
+	validateUpdateResource validate
+}
+
+func (h handler) handleValidation(object, oldObject runtime.RawExtension, newEntity newEntity, validate validate) admission.Response {
+	obj := newEntity()
+	if err := h.decoder.DecodeRaw(object, obj); err != nil {
+		h.logger.Error(err, "could not decode object", "object", object)
+		return admission.Errored(http.StatusUnprocessableEntity, fmt.Errorf("could not decode object %v: %w", object, err))
+	}
+
+	h.logger.Info(fmt.Sprintf("handle validation for %s", obj.GetName()))
```

Comment:
Please include the object type and its full (namespaced) name in the log message, something like:
```suggestion
	h.logger.Info(fmt.Sprintf("validating %s %s", obj.GetKind(), kutil.ObjectName(obj))
```

Reply:
Nit: The more idiomatic way is to benefit from the key/value pairs that the logger accept as context:

```suggestion
	h.logger.Info("Validating resource", "kind", obj.GetKind(), "object", kutil.ObjectName(obj))
```